### DISCUSSION TITLE
vercel: Support for on-demand static reganaration (#1723bypassToken), how can I make use of it?

### DISCUSSION THREAD

**Discussion by TangsWang** [nuxt/nuxt#23847] [Questions]
### Environment

Great job, just what I need, but I have a question: how can I make use of it? (nitro https://github.com/nuxt/nuxt/pull/1723)

### Reproduction

Great job, just what I need, but I have a question: how can I make use of it? (nitro https://github.com/nuxt/nuxt/pull/1723)

### Describe the bug

Great job, just what I need, but I have a question: how can I make use of it? (nitro https://github.com/nuxt/nuxt/pull/1723)

**Comment by pi0** [comment]
Have you read [the docs](https://nitro.unjs.io/deploy/providers/vercel#on-demand-incremental-static-regeneration-isr)?

**Reply by AlexVFornazieri** [reply]
It supposed to work with prerender: true routes in routesRules or it's another context?
After the implementation I try do request to prerendered route with `x-prerender-revalidate` header, and dont't had success, cache keep on it

**Comment by TangsWang** [comment]
@pi0 Thank you very much, but according to the instructions in this document, it requires manual triggering? My requirement is to automatically not cache a page when it returns a 403 status code. This is to prevent illegal web scrapers from affecting legitimate ones. Is there any solution? Thank you.

**Reply by pi0** [reply]
I think if you leverage the nitro native caching layer (not ISR), this is the behavior that only 200 responses will get cached.

ISR is a proprietary feature implemented by Vercel. I am not sure if they do cache 403 responses as well or not but if they do, unless vercel exposes an option, there is nothing much we can do.

**Reply by TangsWang** [reply]
Thank you for your help. However, I'm not sure if Nitro's cache layer will work effectively on Vercel due to its serverless mechanism. Perhaps I should use Redis or something else to persist Nitro's cache layer? Well, this might be too complicated for me. I'm looking for a simpler solution, and I found Next.js's approach on Vercel: https://vercel.com/docs/incremental-static-regeneration/quickstart#on-demand-revalidation. This approach can meet my needs. I'm not sure if Nuxt or Nitro has similar plans.

```
import { revalidatePath } from 'next/cache';
 
export async function GET(request: Request) {
  const { searchParams } = new URL(request.url);
  if (searchParams?.secret !== process.env.MY_SECRET_TOKEN) {
    return new Response(`Invalid credentials`, {
      status: 500,
    });
  }
  revalidatePath('/blog-posts');
  return new Response(
    {
      revalidated: true,
      now: Date.now(),
    },
    {
      status: 200,
    },
  );
}
```
